src/PrintProvider.jsx

```diff
@@ -1,125 +1,129 @@
-    this.hasSingle = true;
-  }
+    hasSingle.current = true;
+  };
 
-  unhideAll() {
+  const unhideAll = () =>  {
     document.body.classList.remove(s.hiddenAll);
-    this.hasSingle = false;
-  }
+    hasSingle.current = false;
+  };
 
-  regPrintable(key, node) {
-    const loose = this.props.loose || this.props.invert;
+  const regPrintable = (key, node) => {
+    const loose = props.loose || props.invert;
     const isSingle = node.props.single || node.props.main;
-    const { hasSingle } = this;
 
     debug('reg printable', key, node);
 
-    if (this.printableRegistry[key] !== undefined || loose) return;
-    setTimeout(() => this.setState({
-      printableNodes: this.state.printableNodes.concat(node),
-    }), 0);
-    this.printableRegistry[key] = this.state.printableNodes.length;
+    if (printableRegistry.current[key] !== undefined || loose) return;
+    setTimeout(
+      () =>
+        setPrintableNodes(printableNodes.concat(node)),
+      0
+    );
+    printableRegistry.current[key] = printableNodes.length;
 
     if (isSingle && !hasSingle) {
-      this.hideAll();
+      hideAll();
     } else if (isSingle) {
-      console.warn(new Error('react-easy-print warning \n\t you\'re using more than one `single` Print component'));
+      console.warn(
+        new Error(
+          'react-easy-print warning \n\t you\'re using more than one `single` Print component'
+        )
+      );
     }
-  }
-
-  unregPrintable(key, isSingle) {
-    const loose = this.props.loose || this.props.invert;
-    if (this.printableRegistry[key] === undefined || this.state.isInPrintPreview || loose) return;
-    this.setState({
-      printableNodes: spliced(this.state.printableNodes, this.printableRegistry[key]),
-    });
-    this.printableRegistry = Object.assign({}, this.printableRegistry, { [key]: undefined });
+  };
+
+  const unregPrintable = (key, isSingle) => {
+    const loose = props.loose || props.invert;
+    if (
+      printableRegistry.current[key] === undefined ||
+      isInPrintPreview ||
+      loose
+    )
+      return;
+    setPrintableNodes(
+      spliced(
+        printableNodes,
+        printableRegistry.current[key]
+      )
+    );
+    printableRegistry.current = {
+      ...printableRegistry.current,
+      [key]: undefined
+    };
     if (isSingle) {
-      this.unhideAll();
+      unhideAll();
     }
-  }
-
-  createRender(children) {
-    const el = document.createElement('div');
-    el.id = 'render';
-    document.body.appendChild(el);
+  };
 
-    ReactDOM.render(<div className={s.printRender} >{children}</div>, el);
-  }
-
-  deleteRender() {
-    const el = document.getElementById('render');
-    el && document.body.removeChild(el);
-  }
-
-  render() {
-    const { isInPrintPreview, printableNodes } = this.state;
-    const loose = this.props.loose || this.props.invert;
-    const { hasSingle } = this;
+  useEffect(() => {
+    window.matchMedia('print').onchange = () => {
```

Comment:
1. `useEffect(() => {...}, []);` is need here too, as I see
2. it's better to keep hooks at function's top